common/thorhelper/thorcommon.cpp

```diff
@@ -1276,6 +1279,31 @@ class CRowStreamReader : public CSimpleInterfaceOf<IExtRowStream>
             filterRow = new RtlDynRow(*actual);
         }
     }
+
+    //interface IVirtualFieldCallback
+    virtual const char * queryLogicalFilename(const void * row) override
+    {
+        if (fieldCallback)
+            return fieldCallback->queryLogicalFilename(row);
+        return "";
+    }
+    virtual unsigned __int64 getFilePosition(const void * row) override
+    {
+        //This is a bit strange - I'm not sure this structure is really right.
+        //There are quite a few layers with file part handlers, stream readers etc.  They possibly should be combined for clarity.
+        if (fieldCallback)
+            return fieldCallback->getFilePosition(row);
+        unsigned __int64 baseOffset = 0; // MORE
+        return prefetchBuffer.tell() + baseOffset;
+    }
+    virtual unsigned __int64 getLocalFilePosition(const void * row) override
+    {
+        if (fieldCallback)
+            return fieldCallback->getLocalFilePosition(row);
+        unsigned part = 0; //MORE
```

Comment:
I think the part will need passing in, or setting into CRowStreamReader, from the readers. e.g. from diskread which reads/creates CRowStreamReader per part.